packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -407,3 +431,251 @@ abstract class InlineSpan extends DiagnosticableTree {
+  ///
+  /// Setting this value to true applies lineThrough and setting it to false
+  /// disables lineThrough.
+  final bool? lineThrough;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationColor]
+  /// if set to non-null.
+  final ui.Color? decorationColor;
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationStyle]
+  /// if set to non-null.
+  final ui.TextDecorationStyle? decorationStyle;
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationThickness]
+  /// if set to non-null.
+  final double? decorationThickness;
+
+  /// An attribute which overwrites [TextSpan.recognizer] if set to non-null.
+  ///
+  /// When a recognizer is specified, the [InlineSpan.updateAttributes] method
+  /// replaces existing recognizers (if any) within the given range with the
+  /// specified recognizer.
+  ///
+  /// When this value is set to [remove], the [InlineSpan.updateAttributes]
+  /// method sets [TextSpan.recognizer] to null in the given range of the returned
+  /// [TextSpan].
+  final RemovableInlineSpanAttribute<GestureRecognizer>? recognizer;
+
+  /// An attribute which overwrites [TextSpan.mouseCursor] if set to non-null.
+  final MouseCursor? mouseCursor;
+
+  /// An attribute which overwrites [TextSpan.onEnter] if set to non-null.
+  ///
+  /// When this value is set to [remove], the [InlineSpan.updateAttributes]
+  /// method sets [TextSpan.onEnter] to null in the given range of the returned
+  /// [TextSpan].
+  final RemovableInlineSpanAttribute<PointerEnterEventListener>? onEnter;
+
+  /// An attribute which overwrites [TextSpan.onExit] if set to non-null.
+  ///
+  /// When this value is set to [remove], the [InlineSpan.updateAttributes]
+  /// method sets [TextSpan.onExit] to null in the given range of the returned
+  /// [TextSpan].
+  final RemovableInlineSpanAttribute<PointerExitEventListener>? onExit;
+
+  /// An attribute which overwrites [TextSpan.spellOut] if set to non-null.
+  final bool? spellOut;
+
+  /// Unsets the given [RemovableInlineSpanAttribute] from an [InlineSpan].
+  ///
+  /// [InlineSpan.updateAttributes] does not mutate the target [InlineSpan].
+  /// Rather, when [remove] is specified for a [RemovableInlineSpanAttribute],
+  /// [InlineSpan.updateAttributes] returns a new [InlineSpan] with that
+  /// attribute unset.
+  static const RemovableInlineSpanAttribute<Never> remove = Right<Never, void>(null);
+
+  static L? _maybeLeft<L extends Object>(Either<L, Object?>? value) => value is Left<L, Object?> ? value.value : null;
+  static R? _maybeRight<R extends Object>(Either<Object?, R>? value) => value is Right<Object?, R> ? value.value : null;
+
+  /// A convenience method, similar to [TextStyle.merge], that merges the
+  /// `textStyle` argument with the [TextStyle] related attributes in this
+  /// [InlineSpanAttributes] object.
+  TextStyle? updateTextStyle(TextStyle? textStyle) {
```

Comment:
Same comment as above, update seems to indicate an in-place operation. Also, should this have the `useResult` annotation?